meteor/client/ui/Settings/BlueprintConfigManifest/index.tsx

```diff
@@ -0,0 +1,225 @@
+
+	layerMappings?: { [studioId: string]: MappingsExt }
+	sourceLayers?: Array<SourceLayerDropdownOption>
+
+	subPanel?: boolean
+
+	configObject: ObjectWithOverrides<IBlueprintConfig>
+	saveOverrides: (newOps: SomeObjectOverrideOp[]) => void
+	pushOverride: (newOp: SomeObjectOverrideOp) => void
+}
+
+/**
+ * Compile a sorted array of all the items currently in the ObjectWithOverrides, and those that have been deleted
+ * @param manifest The config manifest describing all possible items
+ * @param rawConfig The ObjectWithOverrides to look at
+ * @returns Sorted items, with sorted deleted items at the end
+ */
+function getAllCurrentAndDeletedItemsFromOverrides(
+	manifest: ConfigManifestEntry[],
+	rawConfig: ReadonlyDeep<ObjectWithOverrides<IBlueprintConfig>>
+): Array<WrappedOverridableItem<any> & WrappedOverridableExt> {
+	const resolvedObject = applyAndValidateOverrides(rawConfig).obj
+
+	// Convert the items into an array
+	const validItems: Array<WrappedOverridableItemNormal<any> & WrappedOverridableExt> = []
+	for (const entry of manifest) {
+		const value = objectPathGet(resolvedObject, entry.id)
+		// Only include the ones with values or if they are 'required'
+		if (value === undefined && !entry.required) continue
+
+		validItems.push(
+			literal<WrappedOverridableItemNormal<any> & WrappedOverridableExt>({
+				type: 'normal',
+				id: entry.id,
+				computed: value,
+				defaults: objectPathGet(rawConfig.defaults, entry.id),
+				overrideOps: filterOverrideOpsForPrefix(rawConfig.overrides, entry.id).opsForPrefix,
+				manifest: entry,
+			})
+		)
+	}
+
+	const removedOutputLayers: Array<WrappedOverridableItemDeleted<any> & WrappedOverridableExt> = []
+
+	// Find the items which have been deleted with an override
+	const computedOutputLayerIds = new Set(validItems.map((l) => l.id))
+	for (const entry of manifest) {
+		const value = objectPathGet(rawConfig.defaults, entry.id)
+		if (!computedOutputLayerIds.has(entry.id) && value !== undefined) {
+			removedOutputLayers.push(
+				literal<WrappedOverridableItemDeleted<any> & WrappedOverridableExt>({
+					type: 'deleted',
+					id: entry.id,
+					computed: undefined,
+					defaults: value,
+					overrideOps: filterOverrideOpsForPrefix(rawConfig.overrides, entry.id).opsForPrefix,
+					manifest: entry,
+				})
+			)
+		}
+	}
+
+	return [...validItems, ...removedOutputLayers]
+}
+
+export function BlueprintConfigManifestSettings({
+	configManifestId,
+	manifest,
+	alternateConfig,
+	layerMappings,
+	sourceLayers,
+	subPanel,
+
+	configObject,
+	saveOverrides,
+	pushOverride,
+}: IConfigManifestSettingsProps) {
+	const { t } = useTranslation()
+
+	const addRef = useRef<AddItemModalRef>(null)
+	const deleteRef = useRef<DeleteItemModalRef>(null)
+
+	const addItem = useCallback(() => {
+		if (addRef.current) {
+			addRef.current.show()
+		}
+	}, [])
+	const showDelete = useCallback((item: ConfigManifestEntry) => {
+		if (deleteRef.current) {
+			deleteRef.current.show(item)
```

Comment:
This is a bit of an unusual setup, I guess either this could just be component state (i.e. "deleteDialogItem") where there would be no need for this imperative method, or this could be just using `doModalDialog`, which allows an imperative style and accepts callbacks for user decisions.

Reply:
I don't know why I did that model this way, I've changed it to doModalDialog.